### DISCUSSION TITLE
✅ [SOLVED] VIM 9.0 - lsp-hover scrolling (+1, +3) (patch: 9baef1f1) - Popup does not scroll down completely.

### DISCUSSION THREAD

**Discussion by curedent** [prabirshrestha/vim-lsp#1555] [General]
Hi there,

I've experienced some problems with the lsp-hover on VIM 9.0 while scrolling completely down. Often, the scroll is stuck before the end, leaving some precious documentation unreadable. It becomes more apparent when the popup width is smaller.

I spotted the commit ([9baef1f1](https://github.com/prabirshrestha/vim-lsp/pull/1376/commits/5ff5f7a21d9b49b16836ff6275b816fccd60049f#diff-7a7802300eab7236ef211d28deadefe94ead0f793038b9d746719ebe3407aeb6)) by @hrsh7th at the date of 22-12-03 changing the following 
File: /vim-lsp/autoload/vital/_lsp/VS/Vim/Window.vim
Line: 109
```
- let l:topline = min([l:topline, line('$') - l:wininfo.height + 3])
+ let l:topline = min([l:topline, line('$') - l:wininfo.height + 1])
```

If I understand correctly, 

topline = Number of line in the popup, minus the height of the popup, which contains +2 lines for the borders (top, bottom lines), +1 for rounding. So...

```
line('$') - (l:wininfo.height-2) + 1
```

[The initial +3 fix my situation.](https://github.com/prabirshrestha/vim-lsp/pull/1257/commits/b8c0ab1b45e47217b61d2e54977ae88734bdde45#diff-7a7802300eab7236ef211d28deadefe94ead0f793038b9d746719ebe3407aeb6)

I am asking why this line has changed from +3 to +1 ? It might be for NeoVim? 

Thank you!

My VIM version:
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled May 04 2023 10:24:44)

**Comment by mattn** [comment]
Probably, the gap of 1 & 3 is width of border lines.

**Reply by curedent** [reply]
I know, this is what I am saying... 

Why did it change from +3 to +1, where the +3 was working fine?

If there is no apparent reasons, I will just make a patch to be included to fix the issue.

**Reply by prabirshrestha** [reply]
might be @hrsh7th can comment.

The original repo for the library can be found at https://github.com/hrsh7th/vim-vital-vs

**Reply by hrsh7th** [reply]
Hm... I forgotten the reason of this...

I guess I should add `border_width` to `s:info` and then calculate border-offset via this.

I'll check it later.

**Reply by hrsh7th** [reply]
@curedent Could you re-try with patching this diff?

https://github.com/hrsh7th/vim-vital-vs/pull/25/files#diff-efec83c7ca1a82f3135273b518ac444e327bb4cd95fdb0695cf9c35daa92cef4

**Reply by curedent** [reply]
@hrsh7th 

I tested with VIM only and it is working well!

Thank you a lot!

**Reply by hrsh7th** [reply]
Sent a PR.
https://github.com/prabirshrestha/vim-lsp/pull/1556/files
